src/parse.rs

```diff
@@ -532,12 +630,67 @@ impl<'t, 'l, 'py> Parser<'t, 'l, 'py> {
                         token.at.0 + START_TAG_LEN,
                     )?
                     .into(),
-                TokenType::Tag => self.parse_tag(token.content(self.template), token.at)?,
-            })
+                TokenType::Tag => match self.parse_tag(token.content(self.template), token.at)? {
+                    Either::Left(token_tree) => token_tree,
+                    Either::Right(end_tag) => {
```

Comment:
So parse_tag returns `Result<Either<TokenTree, EndTag>>`, `Either<a, b>` seems to work a bit like a tuple, in this instance. So my question, what is the advantage of this compared to matching with a tuple?

Reply:
I think a tuple here would be like using a struct, but `Either` is an enum.